pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java

```diff
@@ -1262,14 +1262,12 @@ public void addTable(TableConfig tableConfig)
     String tableNameWithType = tableConfig.getTableName();
     SegmentsValidationAndRetentionConfig segmentsConfig = tableConfig.getValidationConfig();
 
+    if (getTableConfig(tableNameWithType) != null) {
```

Comment:
(minor) Suggest moving this as the first step of table creation

Reply:
will wait for CI & @siddharthteotia's comment before making this final adjustment